servers/rendering/renderer_rd/renderer_storage_rd.cpp

```diff
@@ -10028,7 +10028,7 @@ RendererStorageRD::RendererStorageRD() {
 
 		actions.renames["COLOR"] = "PARTICLE.color";
 		actions.renames["VELOCITY"] = "PARTICLE.velocity";
-		//actions.renames["MASS"] = "mass"; ?
+		actions.renames["MASS"] = "mass";
```

Comment:
Is this actually used? I just see a local variable which is not used:
```
servers/rendering/renderer_rd/shaders/particles.glsl
240:    float mass = 1.0;
```

Reply:
I think that it is reserved for the attractors.

Reply:
I think we need to figure out why this was commented out with a `?`, and whether attractors in 4.0 are supposed to have a `MASS` factor.

In `3.x` it's used this way:
```
$ rg -g'*.glsl' mass
drivers/gles3/shaders/particles.glsl
81:     float mass = 1.0;
192:                                    force += attractors[i].strength * rel_vec * attenuation * mass;
194:                                    force += attractors[i].strength * attractors[i].dir * attenuation * mass;
```
And there's indeed vertex shader code injected in-between so one can set `MASS` to something different from one to influence it.

In `master` there's also code for attractors, but they don't use use `mass` for now. Either they should be changed to use it, or the built-in should be removed.

Reply:
I could remove this change from the PR, and leave only a fix to INDEX for now.